frontend/src/components/SyncButton.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <div
+    v-if="!isLogin && unsyncedDrives.length"
+    class="col-2 button-container"
+  >
+    <div v-if="inDrivesView">
+      <div v-if="syncStatus === 'untouched'">
+        <svg
+          class="icon"
+          @click="triggerSync"
+        >
+          <use xlink:href="../assets/sprite.svg#sync-alt" />
+        </svg>
+      </div>
+      <div v-else-if="syncStatus === 'loading'">
+        <div class="spinner-border icon" />
+      </div>
+      <div v-else-if="syncStatus === 'success'">
+        <svg class="icon success">
+          <use xlink:href="../assets/sprite.svg#success" />
+        </svg>
+      </div>
+      <div v-else-if="syncStatus === 'fail'">
+        <svg class="icon fail">
+          <use
+            xlink:href="../assets/sprite.svg#fail"
+            href="../assets/sprite.svg#fail"
+          />
+        </svg>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { IS_ONLINE, SYNC, UNSYNCHRONISED_DRIVES } from '@/store/constants';
+import { mapActions, mapState, mapGetters } from 'vuex';
+import { driveListRoute, loginRoute } from '@/router/routes';
+
+export default {
+  name: 'SyncButton',
+  computed: {
+    ...mapState([IS_ONLINE]),
+    ...mapGetters({ unsyncedDrives: UNSYNCHRONISED_DRIVES }),
+  },
+  data() {
+    return {
+      isLogin: this.$router.currentRoute.path === loginRoute.path,
```

Comment:
```suggestion
      inLoginView: this.$router.currentRoute.path === loginRoute.path,
```
nitpick: you're using different naming conventions for the same thing.
`isLogin` and `inDrivesView` do pretty much the same thing, but are named differently, why?

double nitpick: if that convention was to be followed then the `inLoginView` should be named `inDriveListView` :P

Reply:
You are right. The `inDrivesView` condition is logically sufficient. I also refactored it to `isDrivesList` to fit `isLogin` because it was preceeding convention and I should accord to it :) TBC (to be comitted)